UNIXFSv1.md

```diff
@@ -0,0 +1,487 @@
+	repeated uint64 blocksizes = 4;
+	optional uint64 hashType = 5;
+	optional uint64 fanout = 6;
+	optional uint32 mode = 7;
+	optional UnixTime mtime = 8;
+}
+
+message Metadata {
+	optional string MimeType = 1;
+}
+
+message UnixTime {
+	required int64 Seconds = 1;
+	optional fixed32 FractionalNanoseconds = 2;
+}
+```
+
+### IPLD `dag-pb`
+
+A very important other spec for unixfs is the [`dag-pb`](https://ipld.io/specs/codecs/dag-pb/spec/) IPLD spec:
+
+```protobuf
+message PBLink {
+  // binary CID (with no multibase prefix) of the target object
+  optional bytes Hash = 1;
+
+  // UTF-8 string name
+  optional string Name = 2;
+
+  // cumulative size of target object
+  optional uint64 Tsize = 3; // also known as dagsize
+}
+
+message PBNode {
+  // refs to other objects
+  repeated PBLink Links = 2;
+
+  // opaque user data
+  optional bytes Data = 1;
+}
+```
+
+The two different schemas plays together and it is important to understand their different effect,
+- `dag-pb` also named `PBNode` is the "outside" protobuf message, it is the first one you decode. It contain the list of links and some "opaque user data".
+- The `Data` message is the "inside" protobuf message, this can be decoded by first decoding the `PBNode` object and then decoding `Data` from the bytes inside `PBNode.Data` field, this will contain all the rest of information.
+
+This mean we deal with protobuf inside protobuf. This is for history reasons.
+This spec sometimes show this a one object for clarity reasons, when that happen it is implied that the `PBNode.Data` field is encoded into prototbuf.
+
+### Glossary
+
+- Node, Block
+  A node is a word from graph theory, this is the smallest unit present in the graph.
+  Due to how unixfs work, there is a 1 to 1 mapping between nodes and blocks.
+- File
+  A file is some container over an arbitrary sized amounts of bytes.
+  Files can be said to be single block, or multi block, in the later case they are the concatenation of multiple children files.
+- Directory, Folder
+  This is a named collection of children nodes.
+- HAMT Directory
+  This is a Hashed-Array-Mapped-Tree datastructure representing a Directory, thoses are used to split directories into multiple blocks when they get too big.
+- Symlink
+  This represent a POSIX Symlink.<!--TODO: Add link to POSIX spec.-->
+
+### Paths
+
+Path is a collection of path component (some bytes) seperated by `/` 0x2f, red from left to right, it is inspired by POSIX paths.
+
+Components MUST NOT contain `/` characters because else it would break the path into two components.
+
+Components SHOULD be UTF8 unicode.
+
+### Escaping
+
+The `\` may be supposed to trigger an escape sequence.
+
+This might be a thing, but is broken and inconsistent current implementations.
+So until we agree on a new spec for this, you SHOULD NOT use any escape sequence and non ascii character.
+
+
+### SHOULD NOT names
+
+Thoses names SHOULD NOT<!--MUST NOT ? in future revisions--> be used:
```

Comment:
the `<!-- -->` blocks do not get rendered (they are HTML comments), I prefer leaving them in place as they are only usefull for me discussing with reviewers and will be ignored when reading the spec.

Reply:
Also I call them name because they are most not be in the `node.Links[].Name` field, it's actually fine to use `.` and `..` in paths, but not in names of childrens.